extensions/csv/src/test/java/io/deephaven/csv/CsvTest.java

```diff
@@ -1,529 +1,638 @@
-                .row(null)
-                .row(TIMESTAMP_B.toEpochMilli() * 1000L)
-                .newTable();
-        return new CsvTest("timestampMixed", "timestamp-mixed.csv", CsvSpecs.csv(), expected);
-    }
-
-    public static CsvTest timestampLegacy() {
-        final NewTable expected = ColumnHeader.ofInstant("Timestamp")
-                .row(TIMESTAMP_A)
-                .row(null)
-                .row(TIMESTAMP_B)
-                .newTable();
-        return new CsvTest("timestampLegacy", "timestamp-legacy.csv", CsvSpecs.csv(), expected);
-    }
-
-    public static CsvTest bools() {
-        final NewTable expected = ColumnHeader.ofBoolean("Bool")
-                .row(true)
-                .row(null)
-                .row(false)
-                .row(true)
-                .row(false)
-                .row(true)
-                .row(false)
-                .newTable();
-        return new CsvTest("bools", "bools.csv", CsvSpecs.csv(), expected);
-    }
-
-    public static CsvTest chars() {
-        final NewTable expected = ColumnHeader.ofChar("Char")
-                .row('A')
-                .row(null)
-                .row('B')
-                .row('C')
-                .row('1')
-                .row('2')
-                .row('3')
-                .newTable();
-        return new CsvTest("chars", "chars.csv", CsvSpecs.csv(), expected);
-    }
-
-    public static CsvTest byteViaHeader() {
-        final NewTable expected = ColumnHeader.ofByte("Byte")
-                .row((byte) (Byte.MIN_VALUE + 1))
-                .row(null)
-                .row(Byte.MAX_VALUE)
-                .newTable();
-        return new CsvTest("byteViaHeader", "byte.csv", CsvSpecs.builder().header(expected.header()).build(), expected);
-    }
-
-    public static CsvTest byteViaInference() {
-        final NewTable expected = ColumnHeader.ofByte("Byte")
-                .row((byte) (Byte.MIN_VALUE + 1))
-                .row(null)
-                .row(Byte.MAX_VALUE)
-                .newTable();
-        return new CsvTest("byteViaInference", "byte.csv",
-                CsvSpecs.builder().inference(InferenceSpecs.builder().addParsers(Parser.BYTE).build()).build(),
+    @Test
+    public void timestampMillis() {
+        final String input = "" +
+                "Timestamp\n" +
+                "1632769200000\n" +
+                "\n" +
+                "1632772800000\n";
+
+        final Table expected = TableTools.newTable(
+                TableTools.col("Timestamp", DATETIME_A, null, DATETIME_B)
+        );
+
+        invokeTest(input,
+                CsvSpecs.builder().inference(InferenceSpecs.milliTimes()).build(),
                 expected);
     }
 
-    public static CsvTest byteIsShort() {
+    @Test
+    public void timestampMicros() {
+        final String input = "" +
+                "Timestamp\n" +
+                "1632769200000000\n" +
+                "\n" +
+                "1632772800000000\n";
+
```

Comment:
Why were all the resource files removed? I think we should prefer to keep .csv resource files, as it's a very easy way to add new tests.

Reply:
I was finding it very inconvenient to have "expected" and "actual", both with 5 or less lines of data, be in distant parts of the source tree. I found it to be a lot more readable for them to be right next to each other. And these tests can share the same "expected" or "actual" by using a shared string.